File: 2748-number-of-beautiful-pairs/2748-number-of-beautiful-pairs.rs
impl Solution {
    pub fn count_beautiful_pairs(nums: Vec<i32>) -> i32 {
        let mut count = 0;

        for i in 0..nums.len() {
            for j in (i + 1)..nums.len() {
                let s = nums[i].to_string();
                let st = nums[j].to_string();
                if Self::gcd(s.chars().next().unwrap() as i32 - '0' as i32, st.chars().last().unwrap() as i32 - '0' as i32) == 1 {
                    count += 1;
                }
            }
        }

        count
    }

    fn gcd(x: i32, y: i32) -> i32 {
        if y == 0 {
            x
        } else {
            Self::gcd(y, x % y)
        }
    }
}

File: 2748-number-of-beautiful-pairs/2748-number-of-beautiful-pairs.py
import math
from typing import List


class Solution:
    def countBeautifulPairs(self, nums: List[int]) -> int:
        count = 0

        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if self.isCoprime(int(str(nums[i])[0]), int(str(nums[j])[-1])):
                    count += 1

        return count

    def isCoprime(self, x, y) -> bool:
        return math.gcd(x, y) == 1
